lib/components/Table.js

```diff
@@ -40,20 +40,28 @@ const Table = ({
   const calculateTableContainerHeight = () =>
     containerHeight - TABLE_STICKY_HEADER_HEIGHT - TABLE_PAGINATION_HEIGHT;
 
+  const ref = useRef(null);
+
   useEffect(() => {
-    parentContainerRef &&
-      setContainerHeight(parentContainerRef?.current?.clientHeight);
-  }, [parentContainerRef]);
+    let resizeObserver;
+    if (ref.current?.parentNode) {
+      resizeObserver = new ResizeObserver(() => {
+        ref.current && setContainerHeight(ref.current?.parentNode.clientHeight);
+      }).observe(ref.current?.parentNode);
+    }
+    return () => resizeObserver?.disconnect();
+  }, [ref.current?.parentNode]);
 
   if (rowData.every((item) => item.id)) {
     rowData = rowData.map((row) => ({
       ...row,
-      key: useId(row.id),
+      key: row.id,
```

Comment:
@edwinbbu I had to remove this since this is a hook and it is crashing the component when the `rowData` is updated due to difference in the number of hooks executed on rerender.

Reply:
True. I had found a better solution. Can you rebase and update the PR according to https://github.com/bigbinary/neeto-ui/pull/666.

Reply:
@ajmaln _a Can you make this change. The rest looks good.